### PR TITLE
Add r_str_escape_r2 helper function ##util

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by ret2libc — CHANGES_REQUESTED** [review]
A lot of unit tests please

File: libr/util/str.c [hunk 1]
```diff
@@ -1351,6 +1351,32 @@ R_API char *r_str_escape(const char *buf) {
...
+	const char *p = buf;
+	char *q = new_buf;
+	while (*p) {
+		switch (*p) {
+		case ';':
+		case '$':
+		case '`':
+		case '\\':
+		case '"':
+			*q++ = ' ';
```
Comment (thestr4ng3r) [inline libr/util/str.c:1369]: That's not really escaping, it's sanitizing.

File: libr/util/str.c [hunk 2]
```diff
@@ -1351,6 +1351,32 @@ R_API char *r_str_escape(const char *buf) {
...
+			p++;
+			/* FALLTHRU */
```
Comment (thestr4ng3r) [inline libr/util/str.c:1371]: Why fallthrough? Won't this make `pd 42 $$(!rm -rf /)` into `pd 42 $(!rm -rf /)` then?

File: libr/util/str.c [hunk 3]
```diff
@@ -1351,6 +1351,32 @@ R_API char *r_str_escape(const char *buf) {
...
+		default:
+			*q++ = *p++;
+			break;
+		}
+	}
+	*q = '\0';
+	return new_buf;
+}
```
Comment (XVilka) [inline libr/util/str.c:1379]: Please also have an empty line here.

**Comment by trufae** [comment]
As long as there's no such thing as escaping in the oldshell. maybe this function should take an argument to specify filtering or escaping and use cfg.newshell value for it. what do you think @ret2libc ?

**Comment by ret2libc** [comment]
> As long as there's no such thing as escaping in the oldshell. maybe this function should take an argument to specify filtering or escaping and use cfg.newshell value for it. what do you think @ret2libc ?

I'm not sure it is worth doing too complicated stuff for filter/escaping in oldshell as it was never done and doing it now would be wasted time probably?

**Comment by trufae** [comment]
Thats why im doing only filtering for oldshell. Because there's no way to escape things properly in the current shell. So the only safe way would be to remove those bad chars

File: libr/util/str.c (hunk 2 above)
Comment (trufae) [reply libr/util/str.c]: fixed dunno what i smoked

**Pull request merged**